presto-spark-base/src/main/java/com/facebook/presto/spark/util/PrestoSparkUtils.java

```diff
@@ -248,4 +251,24 @@ public static long computeNextTimeout(long queryCompletionDeadline)
     {
         return scala.reflect.ClassTag$.MODULE$.apply(clazz);
     }
+
+    public static <T> Iterator<T> getNullifyingIterator(List<T> list)
```

Comment:
I guess you can't just call remove() on the iterator because that's a code change in Spark?

Reply:
In theory remove on the iterator for an `ArrayList` is an `O(N)` operation (because it has to shift the "tail"). Although in practice I don't think it is going to be an issue, just being on a safer side (just in case there's a query that produces a list with a number of pages that would make this complexity to create a problem)